src/segment-loader.js

```diff
@@ -402,22 +407,16 @@ export default class SegmentLoader extends videojs.EventTarget {
                                                           buffered);
 
     if (percentBuffered >= 90) {
-      // Retry the buffered calculation with the next segment if there is another
-      // segment after the currently selected segment
-      if (mediaIndex + 1 < playlist.segments.length) {
-        percentBuffered = this.getSegmentBufferedPercent_(playlist,
-                                                          mediaIndex + 1,
-                                                          currentTime,
-                                                          buffered);
-      }
+      // Increment the timeCorrection_ variable to push the fetcher forward
+      // in time and hopefully skip any gaps or flaws in our understanding
+      // of the media
+      this.incrementTimeCorrection_(playlist.targetDuration);
```

Comment:
This function was intended to be non-side-effecting. Is there a way to reorganize the responsibilities here so that the time correction can be performed in one of the explicitly side-effecting portions of the code?